Spatial.gd

```diff
@@ -647,7 +652,26 @@ func gen_new_icon(position: Vector3, texture_path: String, waypoint_icon, catego
 		new_icon.visible = is_category_visible(category_item)
 	else:
 		new_icon.visible = false
-	icons.add_child(new_icon)
+	category.add_child(new_icon)
+
+func search_node_by_name(node, target_name):
+	for child in node.get_children():
+		var result = _search_node_by_name(child, target_name)
+		if result:
+			return result
+
+func _search_node_by_name(node: Spatial, target_name: String):
+	var split_name = target_name.split(".")
```

Comment:
We have been trying to move away from the stringy representation of "a.b.c" for a while now. Lets not build new core infrastructure that relies on it.